cmd/kubeadm/app/cmd/init.go

```diff
@@ -342,6 +360,14 @@ func newInitData(cmd *cobra.Command, args []string, options *initOptions, out io
 		}
 	}
 
+	certificateKey := ""
```

Comment:
Considering that the goal of this function is to validate initOptions, IMO it is more appropriate to move the logic about generating a new key to the new `upload-certs` phase;